QuizApp/Core/Custom UI/FinishPopup/FinishPopupController.swift

```diff
@@ -0,0 +1,179 @@
+    init() {
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    @objc func closeButtonTapped() {
+        dismiss(animated: true, completion: nil)
+    }
+    
+    private func setupPopupViewConstraints() {
+        NSLayoutConstraint.activate([
+            popupView.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            popupView.centerYAnchor.constraint(equalTo: view.centerYAnchor),
+            popupView.widthAnchor.constraint(equalToConstant: Constants.PopupView.widthAnchor),
+            popupView.heightAnchor.constraint(equalToConstant: Constants.PopupView.heightAnchor)
+        ])
+    }
+    
+    private func setupSuccessIconConstraints() {
+        NSLayoutConstraint.activate([
+            successIcon.centerXAnchor.constraint(equalTo: popupView.centerXAnchor),
+            successIcon.topAnchor.constraint(equalTo: popupView.topAnchor, constant: Constants.SuccessIcon.topAnchor)
+        ])
+    }
+    
+    private func setupPopupLabelConstraints() {
+        NSLayoutConstraint.activate([
+            popupLabel.centerXAnchor.constraint(equalTo: popupView.centerXAnchor),
+            popupLabel.topAnchor.constraint(equalTo: successIcon.bottomAnchor, constant: Constants.PopupLabel.topAnchor)
+        ])
+    }
+    
+    private func setupPopupScoreLabelConstraints() {
+        NSLayoutConstraint.activate([
+            popupScoreLabel.centerXAnchor.constraint(equalTo: popupView.centerXAnchor),
+            popupScoreLabel.topAnchor.constraint(equalTo: popupLabel.bottomAnchor, constant: Constants.PopupScoreLabel.topAnchor)
+        ])
+    }
+    
+    private func setupDividerViewConstraints() {
+        NSLayoutConstraint.activate([
+            dividerView.topAnchor.constraint(equalTo: popupScoreLabel.bottomAnchor, constant: Constants.DividerView.topAnchor),
+            dividerView.leadingAnchor.constraint(equalTo: popupView.leadingAnchor),
+            dividerView.trailingAnchor.constraint(equalTo: popupView.trailingAnchor),
+            dividerView.heightAnchor.constraint(equalToConstant: Constants.DividerView.heightAnchor)
+        ])
+    }
+    
+    private func setupCloseButtonConstraints() {
+        NSLayoutConstraint.activate([
+            closeButton.widthAnchor.constraint(equalToConstant: Constants.CloseButton.widthAnchor),
+            closeButton.heightAnchor.constraint(equalToConstant: Constants.CloseButton.heightAnchor),
+            closeButton.centerXAnchor.constraint(equalTo: popupView.centerXAnchor),
+            closeButton.topAnchor.constraint(equalTo: dividerView.bottomAnchor, constant: Constants.CloseButton.topAnchor)
+        ])
+    }
+}
+
+extension FinishPopupController {
+    enum Constants {
+        enum PopupView {
+            static let backgroundColor = UIColor(hex: "FFC44A")
+            static let cornerRadius: CGFloat = 10
+            static let widthAnchor: CGFloat = 270
+            static let heightAnchor: CGFloat = 176
+        }
+        enum PopupLabel {
```

Comment:
ეს ძალიან სუბიექტურია, მაგრამ მე enum-ებს შორის გამოვტოვებდი ადგილებს
```suggestion
        }
        
        enum PopupLabel {
```